services/venice-controller/src/main/java/com/linkedin/venice/controller/DeferredVersionSwapService.java

```diff
@@ -0,0 +1,237 @@
+  private final VeniceControllerMultiClusterConfig veniceControllerMultiClusterConfig;
+  private final VeniceParentHelixAdmin veniceParentHelixAdmin;
+  private final ScheduledExecutorService deferredVersionSwapExecutor = Executors.newSingleThreadScheduledExecutor();
+  private final DeferredVersionSwapStats deferredVersionSwapStats;
+  private static final Logger LOGGER = LogManager.getLogger(DeferredVersionSwapService.class);
+
+  public DeferredVersionSwapService(
+      VeniceParentHelixAdmin admin,
+      VeniceControllerMultiClusterConfig multiClusterConfig,
+      DeferredVersionSwapStats deferredVersionSwapStats) {
+    this.veniceParentHelixAdmin = admin;
+    this.allClusters = multiClusterConfig.getClusters();
+    this.veniceControllerMultiClusterConfig = multiClusterConfig;
+    this.deferredVersionSwapStats = deferredVersionSwapStats;
+  }
+
+  @Override
+  public boolean startInner() throws Exception {
+    deferredVersionSwapExecutor.scheduleAtFixedRate(
+        new DeferredVersionSwapTask(),
+        0,
+        veniceControllerMultiClusterConfig.getDeferredVersionSwapSleepMs(),
+        TimeUnit.MILLISECONDS);
+    return true;
+  }
+
+  @Override
+  public void stopInner() throws Exception {
+    stop.set(true);
+    deferredVersionSwapExecutor.shutdown();
+  }
+
+  private Set<String> getRegionsForVersionSwap(Map<String, Integer> candidateRegions, Set<String> targetRegions) {
+    Set<String> remainingRegions = new HashSet<>(candidateRegions.keySet());
+    remainingRegions.removeAll(targetRegions);
+    return remainingRegions;
+  }
+
+  private String getTargetRegion(Set<String> targetRegions) {
+    return targetRegions.iterator().next();
+  }
+
+  private StoreResponse getStoreForRegion(String clusterName, String targetRegion, String storeName) {
+    Map<String, ControllerClient> controllerClientMap =
+        veniceParentHelixAdmin.getVeniceHelixAdmin().getControllerClientMap(clusterName);
+    ControllerClient targetRegionControllerClient = controllerClientMap.get(targetRegion);
+    return targetRegionControllerClient.getStore(storeName);
+  }
+
+  private class DeferredVersionSwapTask implements Runnable {
+    @Override
+    public void run() {
+      while (!stop.get()) {
+        try {
+          for (String cluster: allClusters) {
+            if (!veniceParentHelixAdmin.isLeaderControllerFor(cluster)) {
+              continue;
+            }
+
+            List<Store> stores = veniceParentHelixAdmin.getAllStores(cluster);
+            for (Store store: stores) {
+              if (StringUtils.isEmpty(store.getTargetSwapRegion())) {
+                continue;
+              }
+
+              int targetVersionNum = store.getLargestUsedVersionNumber();
+              Version targetVersion = store.getVersion(targetVersionNum);
+              if (targetVersion == null) {
+                continue;
+              }
+
+              if (targetVersion.getStatus() != VersionStatus.PUSHED) {
```

Comment:
Can you add some comments here explaining why we need to check for PUSHED status and what it means